### PR TITLE
Using publisher data file for publisher exclusion (removing ruleset code)

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: #14498

This removes the code related to caching/using rulesets and its related dependencies)

Depends on https://github.com/brave-intl/bat-client/pull/88/ (will need `bat-client 3.4.0` when available)

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
(Code testing: Be sure to pull `bat-client` from https://github.com/brave-intl/bat-client/pull/88/  and re-install modules before testing)
1. Run Brave from this branch with a clean profile, enable payments
2. Ensure that auto-include is on
3. Pull up `publisher-data.json` for testing data (Note, file is a bit large)
4. Visit sites and satisfy the minimum requirements to add them to the ledger. (for example amazon.com)
5. Ensure that ledger table inclusion matches what is in `publisher-data.json`

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.
- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/browser/api/ledger.js [hunk 1]
```diff
@@ -2133,8 +1988,8 @@ const callback = (err, result, delayTime) => {
 }
 
 const onCallback = (state, result, delayTime) => {
-  let results
-  let entries = client && client.report()
+  // let results
+  // let entries = client && client.report()
```
Comment (jasonrsadler) [inline app/browser/api/ledger.js]: Are these done?

Comment (ryanml) [reply app/browser/api/ledger.js]: removed

File: app/browser/api/ledger.js [hunk 2]
```diff
@@ -726,70 +686,29 @@ const updatePublishers = (state, publisherKeys) => {
...
+const doneP = (err, result, callback) => {
+  callback(err, result)
+}
 
-  const done = (err, result) => {
-    doneP = true
-    callback(err, result)
+const getPublisherExclude = (publisherKey, callback) => {
+  if (!publisherInfoData) {
+    return setTimeout(() => module.exports.getPublisherExclude(publisherKey, callback), 10 * ledgerUtil.milliseconds.second)
```
Comment (NejcZdovc) [inline app/browser/api/ledger.js]: what we will get with 10 sec delay?

File: app/browser/api/ledger.js [hunk 3]
```diff
@@ -726,70 +686,29 @@ const updatePublishers = (state, publisherKeys) => {
...
   }
 
-  if (!v2RulesetDB) {
-    return setTimeout(() => excludeP(publisherKey, callback), 5 * ledgerUtil.milliseconds.second)
+  const publisher = publisherInfoData.find(p => p.first() === publisherKey)
+  // We should include by default
+  if (typeof publisher === 'undefined') {
```
Comment (NejcZdovc) [inline app/browser/api/ledger.js]: we should check default ruleset from client as well

File: app/browser/api/ledger.js [hunk 4]
```diff
@@ -2173,44 +2025,17 @@ const onCallback = (state, result, delayTime) => {
...
-              appActions.onPublisherOptionUpdate(publisherKey, 'exclude', exclude)
-              savePublisherOption(publisherKey, 'exclude', exclude)
-            })
-          }
-        }
-      })
-    }
+    const publishers = ledgerState.getPublishers(state)
+    for (let item of publishers) {
+      const publisherKey = item[0]
```
Comment (NejcZdovc) [inline app/browser/api/ledger.js]: Aren't publishers immutable object?

File: app/browser/api/ledger.js (hunk 2 above)
Comment (ryanml) [reply app/browser/api/ledger.js]: This should probably be ~15 seconds, it's just that `publisherInfoData` may not be yet available due to startup delay

File: app/browser/api/ledger.js (hunk 4 above)
Comment (ryanml) [reply app/browser/api/ledger.js]: I will double check, but this was basically preserved from the previous `if (notInProgress)` block here: https://github.com/brave/browser-laptop/pull/14751/files#diff-3cdfdb124f718787d584e9e2b82fcedcL2198

Comment (ryanml) [reply app/browser/api/ledger.js]: @NejcZdovc you can iterate over a `Map` using `for (let item of map)`

**Comment by ryanml** [comment]
@jasonrsadler this is also ready for another look on your end.

Basic rules from the default ruleset regarding government sites have been implemented in `shouldExclude`. Test have been added for the variety of government domains present in the `bat-publisher` ruleset.

**Comment by ryanml** [comment]
@NejcZdovc `bat-client` upgraded, `package-lock` updated

File: app/browser/api/ledger.js [hunk 5]
```diff
@@ -727,70 +687,61 @@ const updatePublishers = (state, publisherKeys) => {
...
-      let result
-      try {
-        result = JSON.parse(data.value)
-      } catch (ex) {
-        console.error(v2RulesetPath + ' stream invalid JSON ' + data.entry + ': ' + data.value)
-      }
+  if (typeof publisher === 'undefined') {
+    // If publisher doesn't exist in infoData, check the exclusion rules
+    const exclude = module.exports.shouldExclude(publisherKey)
+    return module.exports.doneP(null, exclude, callback)
```
Comment (NejcZdovc) [inline app/browser/api/ledger.js]: nit: can we just rename `exclude` in `result` and have one return?

File: app/browser/api/ledger.js [hunk 6]
```diff
@@ -1079,7 +1031,11 @@ const pageDataChanged = (state, viewData = {}, keepInfo = false) => {
       }
     }
 
-    if (!publisherKey || (ledgerUtil.blockedP(state, publisherKey))) {
+    if (!publisherKey) {
```
Comment (NejcZdovc) [inline app/browser/api/ledger.js]: Why this was added?

File: app/browser/api/ledger.js (hunk 4 above)
Comment (NejcZdovc) [reply app/browser/api/ledger.js]: Yeah I was too fast with this one, I though that you were doing something else, sorry

File: app/browser/api/ledger.js (hunk 5 above)
Comment (ryanml) [reply app/browser/api/ledger.js]: Can do

Comment (ryanml) [reply app/browser/api/ledger.js]: done

File: app/browser/api/ledger.js (hunk 6 above)
Comment (NejcZdovc) [reply app/browser/api/ledger.js]: explained in DM

**Comment by NejcZdovc** [comment]
master 40e451d
0.24 2429dabd7
0.23 e92ba8673

**Pull request merged**